tests/asset-store/testsuite/testdata.go

```diff
@@ -0,0 +1,125 @@
+package testsuite
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/kyma-project/kyma/components/assetstore-controller-manager/pkg/apis/assetstore/v1alpha2"
+	"github.com/kyma-project/kyma/tests/asset-store/pkg/file"
+	"github.com/kyma-project/kyma/tests/asset-store/pkg/upload"
+	"github.com/pkg/errors"
+)
+
+const basePath = "testdata"
+
+type testData struct {
+	url string
+}
+
+func newTestData(url string) *testData {
+	return &testData{url: url}
+}
+
+// TODO: Test private bucket
+func (u *testData) Upload() (*upload.Response, error) {
+	publicFiles, err := u.loadFiles(paths())
+	if err != nil {
+		return nil, errors.Wrapf(err, "while loading files to upload")
+	}
+
+	resp, err := upload.Do("", upload.UploadInput{
+		PrivateFiles: nil,
+		PublicFiles:  publicFiles,
+		Directory:    "",
+	}, u.url)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while uploading files")
+	}
+
+	return resp, nil
+}
+
+func (u *testData) loadFiles(paths []string) ([]*os.File, error) {
+	var files []*os.File
+
+	for _, path := range paths {
+		f, err := file.Open(path)
+		if err != nil {
+			return nil, err
+		}
+
+		files = append(files, f)
+	}
+
+	return files, nil
+}
+
+func paths() []string {
+	return []string{
+		localPath("foo.yaml"),
+		localPath("bar.yaml"),
+		localPath("package.tar.gz"),
+	}
+}
+
+func localPath(fileName string) string {
+	return fmt.Sprintf("%s/%s", basePath, fileName)
+}
+
+func verifyUploadedAsset(files []uploadedFile, shouldExist bool, logFn func(format string, args ...interface{})) error {
```

Comment:
Wouldn't it be clearer to divide this function into two separate ones, because right now it has two separate flows that depend on bool argument and don't seem to share any common logic (do two different things). Just a thought.